File: modules/app-exit/android/src/main/java/expo/modules/appexit/AppExitModule.kt
package expo.modules.appexit

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition
import kotlin.system.exitProcess

class AppExitModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("AppExit")

    Function("exitApp") {
      val activity = appContext.currentActivity
      activity?.finishAffinity()
      exitProcess(0)
    }
  }
}

File: modules/app-exit/ios/AppExitModule.swift
import ExpoModulesCore

public class AppExitModule: Module {
  public func definition() -> ModuleDefinition {
    Name("AppExit")

    Function("exitApp") {
      // Force-quit so an in-progress IPA/APK install can replace this binary.
      exit(0)
    }
  }
}
